tensorflow/core/data/dataset_utils.cc

```diff
@@ -219,17 +219,18 @@ void DefaultOptimizationGraphRewrites(
       optimization_disabled->insert(kShuffleAndRepeatFusionOpt);
     }
   }
-  const bool has_autotune = optimization_options.optional_autotune_case() ==
-                            OptimizationOptions::kAutotune;
+  const auto& autotune_options = options.autotune_options();
```

Comment:
the code will have to assume that the `options` object has been generated by code before your PR and so you will need to keep the old logic + add the new logic for backwards compatibility

this comment applies to all places that are accessing the `Options` object that was passed into a tf.data op

Reply:
okay, will modify